ci/scripts/bundle.py

```diff
@@ -0,0 +1,232 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import io
+import os
+import pathlib
+import re
+
+
+def read_content(path_or_content):
```

Comment:
Of course up to you but I find the behavior of this function surprising. More often than not, when you accept a path argument then accepting a string usually still represents the filepath/name, not the contents itself.

If you wanted you could refactor this to just be:

```python
def read_content(path_or_content: io.StringIO | pathlib.Path | str):
    with open(path_or_content) as f:
        return f.read()
```

For the caller this would require them to pass `io.StringIO("some_contents")` instead of just `"some_contents"`, but I think makes it less ambiguous in case someone expects a string argument here to still represent a path